pkg/parsers/operatingsystem/operatingsystem_linux.go

```diff
@@ -18,15 +23,29 @@ var (
 
 // GetOperatingSystem gets the name of the current operating system.
 func GetOperatingSystem() (string, error) {
-	b, err := ioutil.ReadFile(etcOsRelease)
+	osReleaseFile, err := os.Open(etcOsRelease)
 	if err != nil {
 		return "", err
 	}
-	if i := bytes.Index(b, []byte("PRETTY_NAME")); i >= 0 {
-		b = b[i+13:]
-		return string(b[:bytes.IndexByte(b, '"')]), nil
+	defer osReleaseFile.Close()
+
+	scanner := bufio.NewScanner(osReleaseFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "PRETTY_NAME") {
+			data := strings.SplitN(line, "=", 2)
+			if len(data) == 2 {
+				prettyNames, err := shellwords.Parse(data[1])
+				if err == nil && len(prettyNames) == 1 {
+					return prettyNames[0], nil
+				}
+			}
+			return "", fmt.Errorf("PRETTY_NAME is invalid")
```

Comment:
Can we elaborate on why its invalid?  Although, if we add "=" to the "HasPrefix" up above, I think the `if len(data) == 2` will always be `true` and we'll never get here.
If you're worried about the case of an invalid file where they just have `PRETTY_NAME` on a line all by itself (w/o "=") then you could add a specific check for that:
```
if line == "PRETTY_NAME" {
  return "", fmt.Errorf("PRETTY_NAME property is missing '=value'")
}
```
before line 35.